src/alephbet.coffee

```diff
@@ -67,6 +67,34 @@ class AlephBet
       @storage().get("#{@options.name}:variant")
 
     pick_variant: ->
+      # we are checking that all variants of experiment has weights
+      variants_has_weight = utils.checkWeights(@variants).every (contains_weight) -> contains_weight
+      utils.log("all variants has weight: #{variants_has_weight}")
+      # if all variants has weights than we should fire version for variants with weight
+      if variants_has_weight then @pick_weighted_variant() else @pick_unweighted_variant()
+
+    pick_weighted_variant: ->
+
+      # Choosing a weighted variant:
+      # For A, B, C with weights 10, 30, 60
+      # variants = A, B, C
+      # weights = 10, 30, 60
+      # weightSum = 100
+      # weightedIndex = 21 (random number between 0 and weight sum)
+      # ABBBCCCCCC - (every letter occurence should by multiplied by 10)
+      # =======^
+      # Select C
+
+      # I'm assuming that all weights will sum up to 100
+      # then I pick random number from 0 - 100
+      weightedIndex = Math.floor(Math.random() * 100)
```

Comment:
I believe you should use `_random` instead of `Math.random` so this plays nicely with [user-based tracking](https://github.com/Alephbet/alephbet/wiki/User-based-and-Cross-device-tracking)